pkg/scheduler/internal/queue/scheduling_queue_test.go

```diff
@@ -1870,6 +1870,38 @@ func TestPerPodSchedulingMetrics(t *testing.T) {
 		t.Fatalf("Failed to pop a pod %v", err)
 	}
 	checkPerPodSchedulingMetrics("Attempt twice with update", t, pInfo, 2, timestamp)
+
+	// Case 4: A gated pod is created and scheduled after lifting gate. The queue operations are
```

Comment:
We should be using a test table for these cases or split into separate test functions. That should make tests easier to debug in the future.

This is beyond the scope of this PR, but please consider working on this after this has been merged and cherry-picked.

Although, I would prefer this new test goes into a separate function in this PR.

Reply:
we do have some test code not following the table-driven test style. it can be followed-up with other tests.

Reply:
I agree to follow the table-driven test style. I didn't want to have an out-of-scope change in this PR. I will follow up with another PR to address this comment.